inference-engine/src/readers/ir_reader/ie_ir_parser.cpp

```diff
@@ -46,44 +47,223 @@ IRParser::IRParser(size_t version, const std::vector<InferenceEngine::IExtension
+
+            outputs.push_back(std::make_shared<ngraph::op::util::SubGraphOp::ConcatOutputDescription>
+                    (result_id_in_function[body_result_index],
+                    output_number,
+                    start,
+                    stride,
+                    part_size,
+                    end,
+                    axis));
+        } else {
+            // otherwise create ngraph::TensorIterator::BodyOutput. -1 means last iteration.
+            outputs.push_back(std::make_shared<ngraph::op::util::SubGraphOp::BodyOutputDescription>
+                    (result_id_in_function[body_result_index],
+                    output_number,
+                    -1));
+        }
+        output_number++;
+    }
+    return outputs;
+}
+
+void V10Parser::XmlDeserializer::on_adapter(const std::string& name, ngraph::ValueAccessor<void>& adapter) {
+    std::string val;
+
+    // for TensorIterator look for 'port_map' as 'data' does not exist
+    if (node.child("port_map")) {
+        if (auto a = ngraph::as_type<ngraph::AttributeAdapter<std::vector<std::shared_ptr
+                    <ngraph::op::util::SubGraphOp::InputDescription>>>>(&adapter)) {
+            a->set(parseInputDescription(node));
+        } else if (auto a = ngraph::as_type<ngraph::AttributeAdapter<std::vector<std::shared_ptr
+                    <ngraph::op::util::SubGraphOp::OutputDescription>>>>(&adapter)) {
+            a->set(parseOutputDescription(node));
+        }
+    }
+
+    if (!getStrAttribute(node.child("data"), name, val)) return;
+    if (auto a = ngraph::as_type<ngraph::AttributeAdapter<ngraph::element::Type>>(&adapter)) {
+        static_cast<ngraph::element::Type&>(*a) = details::convertPrecision(val);
+    } else if (auto a = ngraph::as_type<ngraph::AttributeAdapter<ngraph::PartialShape>>(&adapter)) {
+        std::vector<int64_t> shape;
+        std::vector<ngraph::Dimension> dims;
+        if (!getParameters<int64_t>(node.child("data"), name, shape)) return;
+        for (const auto& dim : shape) dims.emplace_back(dim);
+        static_cast<ngraph::PartialShape&>(*a) = ngraph::PartialShape(dims);
+    } else if (auto a = ngraph::as_type<ngraph::AttributeAdapter<ngraph::Shape>>(&adapter)) {
+        std::vector<size_t> shape;
+        if (!getParameters<size_t>(node.child("data"), name, shape)) return;
+        static_cast<ngraph::Shape&>(*a) = ngraph::Shape(shape);
+    } else if (auto a = ngraph::as_type<ngraph::AttributeAdapter<ngraph::Strides>>(&adapter)) {
+        std::vector<size_t> shape;
+        if (!getParameters<size_t>(node.child("data"), name, shape)) return;
+        static_cast<ngraph::Strides&>(*a) = ngraph::Strides(shape);
+#ifdef __APPLE__
+    } else if (auto a = ngraph::as_type<ngraph::AttributeAdapter<std::vector<size_t>>>(&adapter)) {
+        std::vector<size_t> result;
+        if (!getParameters<size_t>(node.child("data"), name, result)) return;
+        static_cast<std::vector<size_t>&>(*a) = result;
```

Comment:
Only this line is different for `__APPLE__` and other. Have you consider move macro her or is this common pattern in our code?

Reply:
It is common approach, as size_t is not uint_64t on OSX.

Reply:
👍